Add logical and op

cccclai opened the pull request:
Summary:
As title, add the logical and op for an internal model

Rollback Plan:

Differential Revision: D80122607

cccclai commented:
Can I get a review on this PR? @haowhsu-quic @shewu-quic @winskuo-quic @DannyYuyang-quic

haowhsu-quic commented on backends/qualcomm/tests/test_qnn_delegate.py:
```diff
@@ -856,6 +856,13 @@ def test_qnn_backend_log(self):
         sample_input = (torch.rand([1, 2, 3, 4]),)
         self.lower_module_and_test_output(module, sample_input)
 
+    def test_qnn_backend_logical_and(self):
```
Could you also add this test case to `TestQNNQuantizedOperator`? thanks.

cccclai replied:
Good catch - updated

haowhsu-quic commented on backends/qualcomm/builders/op_logical_and.py:
```diff
@@ -0,0 +1,66 @@
+            nodes_to_wrappers,
+        )
+
+        logical_not_op = PyQnnWrapper.PyQnnOpWrapper(
```
`logical_and_op` would be better I think.

cccclai replied:
Ah yes, copy paste error...thanks and updated

DannyYuyang-quic commented on backends/qualcomm/builders/qnn_constants.py:
```diff
@@ -200,6 +200,11 @@ class OpElementWiseNeuron:
     param_beta: str = "beta"
 
 
+@dataclass(init=False, frozen=True)
+class OpElementWiseAnd:
+    op_name: str = "ElementWiseAnd"
```
Hi @cccclai, since we already have `OpElementWiseAnd` at line 111, I believe there's no need to add it again. Thanks!!

DannyYuyang-quic reviewed (approved):
Thanks for supporting more ops!!

shewu-quic commented on backends/qualcomm/builders/op_logical_and.py:
```diff
@@ -0,0 +1,66 @@
+
+@register_node_visitor
+class And(NodeVisitor):
+    target = ["aten.logical_and.default"]
```
I think you need to add new target in `quantizer/annotator.py` https://github.com/pytorch/executorch/blob/1520f9f9de6d410894fad7acbd8d9f53924c2a67/backends/qualcomm/quantizer/annotators.py#L831

cccclai replied:
Is the latest version correct? It seems still missing the q/dq ops

cccclai committed:
Add logical and op (#13342)

Summary:

As title, add the logical and op for an internal model

Differential Revision: D80122607
